src/functional/fencil_processors/callables/cpp/build.py

```diff
@@ -0,0 +1,191 @@
+import pathlib
+import subprocess
+import textwrap
+from typing import Dict, Final, Optional, Sequence
+
+from eve import Node
+from eve.codegen import JinjaTemplate, TemplatedGenerator
+from functional.fencil_processors import defs
+
+
+_build_subdir: Final = "build"
+
+
+class FindDependency(Node):
+    name: str
+    version: str
+
+
+class LinkDependency(Node):
+    name: str
+    target: str
+
+
+class CMakeListsFile(Node):
+    project_name: str
+    find_deps: Sequence[FindDependency]
+    link_deps: Sequence[LinkDependency]
+    source_names: Sequence[str]
+    bin_output_suffix: str
+
+
+class CMakeListsGenerator(TemplatedGenerator):
+    CMakeListsFile = JinjaTemplate(
+        """
+        project({{project_name}})
+        cmake_minimum_required(VERSION 3.20.0)
+
+        # Languages
+        enable_language(CXX)
+        set(CMAKE_CXX_STANDARD 17)
+        set(THREADS_PREFER_PTHREAD_FLAG ON)
+        find_package(Threads REQUIRED)
+        link_libraries(Threads::Threads)
+
+        # Paths
+        list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
+        list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})
+
+        set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+        set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+        set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+        # Find dependencies
+        include(FetchContent)
+
+        {{"\\n".join(find_deps)}}
+
+        # Targets
+        add_library({{project_name}} MODULE)
+
+        set_target_properties({{project_name}} PROPERTIES PREFIX "" SUFFIX ".{{bin_output_suffix}}")
+
+        target_sources({{project_name}}
+            PRIVATE
+                {{"\\n".join(source_names)}}
+        )
+
+        # Link dependencies
+        {{"\\n".join(link_deps)}}
+        """
+    )
+
+    def visit_FindDependency(self, dep: FindDependency):
+        match dep.name:
+            case "pybind11":
+                return "find_package(pybind11 CONFIG REQUIRED)"
+            case "gridtools":
+                return textwrap.dedent(
+                    """\
+                    FetchContent_Declare(GridTools
+                        GIT_REPOSITORY https://github.com/GridTools/gridtools.git
+                        GIT_TAG        master
+                    )
+                    FetchContent_MakeAvailable(GridTools)\
+                    """
+                )
+            case "openmp":
+                return "find_package(OpenMP REQUIRED)"
+            case _:
+                raise ValueError("Library {name} is not supported".format(name=dep.name))
+
+    def visit_LinkDependency(self, dep: LinkDependency):
+        match dep.name:
+            case "pybind11":
+                lib_name = "pybind11::module"
+            case "gridtools":
+                lib_name = "GridTools::gridtools"
```

Comment:
Changed it to `fn_naive`. We still need to think a bit about how to structure GPU support on the Python side and how we present that to the user so this will probably do until then.